### PR TITLE
[server-wallet] Add a 2x Wallet createChannel test

### PR DESCRIPTION
**Pull request by geoknee**
Preparatory work towards ledger-funding. Test can cover some or all funding strategies.

- Test fails because 'status' does not update as expected
- Apparently no difference in behaviour when passing a different funding strategy
- test is (almost) self contained with the help of a bit of `execSync` and a new `DBAdmin().migrate()` helper. It just needs a postgres server running
- generated #2659

### CODE REVIEW

File: packages/server-wallet/src/__test__/create-channel-2-wallets.test.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+});
+afterAll(async () => {
+  await Promise.all([a.destroy(), b.destroy()]);
+  execSync('dropdb TEST_A $PSQL_ARGS');
+  execSync('dropdb TEST_B $PSQL_ARGS');
+});
+
+const testCases: FundingStrategy[] = ['Unfunded', 'Direct', 'Ledger', 'Virtual'];
+
+it.each(testCases)(
```
Comment (geoknee) [inline packages/server-wallet/src/__test__/create-channel-2-wallets.test.ts:33]: We may (or may not) wish to assert differently depending on the funding strategy.

Comment (geoknee) [reply packages/server-wallet/src/__test__/create-channel-2-wallets.test.ts:33]: @tomclose we discussed a 'fake' funding strategy. Perhaps this is already covered by 'Unfunded'?

**Comment by snario** [comment]
Added 0329ee7 based on [this Notion strategy](https://www.notion.so/elfour/Server-wallet-thoughts-ba7c8177dae44544987e5956e8233901#3fd8d1a7b2064d328eb8efb10c2a963f). Doesn't include _rejecting_ objectives, and just uses an in-memory objectives store, but gets the test to pass.

**Comment by snario** [comment]
Going to close this in favour of new smaller PRs

**Pull request closed without merging**